core/vtk/ttkDiscreteGradient/ttkDiscreteGradient.cpp

```diff
@@ -101,7 +101,16 @@ int ttkDiscreteGradient::fillCriticalPoints(
     cellIds->SetTuple1(i, critPoints_cellIds[i]);
     cellScalars->SetTuple1(i, scalars[critPoints_PLVertexIdentifiers[i]]);
     isOnBoundary->SetTuple1(i, critPoints_isOnBoundary[i]);
+#ifdef TTK_ENABLE_MPI
+    if(ttk::hasInitializedMPI()) {
```

Comment:
I would argue that this test should happen in the base code.
i.e. when you fill the output vector critPoints_PLVertexIdentifiers, there you should check if you should use local or global ids.

the vtk layer can be completely independent from that.

Reply:
The corresponding code in the base layer is the `DiscreteGradient::setCriticalPoints` method (the first one inside DiscreteGradient_Template.h).

Reply:
`critPoints_PLVertexIdentifiers` in local ids is needed to add elements to the `cellScalars` array. Moving the test to the base layer means adding a new array as argument to setCriticalPoints, as `critPoints_PLVertexIdentifiers` is still needed here. I thought it would be simpler to do this in the vtk layer, as adding arguments only useful in MPI is verbose and makes the code harder to read.
Do you still want me to do this change?

Reply:
understood, let's keep it this way then.
thanks for the clarification.